docs/DeveloperGuide.md

```diff
@@ -1,34 +1,845 @@
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+    1.  Click `Configure` &gt; `Project Defaults` &gt; `Project Structure`
 
-## Non-Functional Requirements
+    2.  Click `New...` and find the directory of the JDK
 
-{Give non-functional requirements}
+4.  Click on `Import Project`
 
-## Glossary
+5.  Locate and select the `build.gradle` file, then click `OK`
+
+6.  Click `Open as Project`
+
+7.  Click `OK` to use the default settings provided
+
+### 1.3. Verifying the Setup
+1.  In an IntelliJ terminal, run `gradlew build`
+
+2.  Move to the folder `build` &gt; `libs` and run: `java -jar atas-2.0.0.jar`
+
+    1.  To use **ATAS**, type a valid command into the terminal and press the enter key to run the command.  
+        e.g. Typing `help` and pressing the enter key will show the available commands
+
+    2.  Some example commands you can try to get familiar with **ATAS**:
+
+        -   `help`: Lists the commands that **ATAS** supports.
+
+        -   `assignment n/Assignment One m/CS2113T d/01/02/20 1600 c/Important Assignment`: Adds an assignment called **Assignment One** for the module **CS2113T**. This assignment is due on **1st February 2020 4pm** and the comments for this assignment is that it is an **Important Assignment**.
+
+        -   `exit`: Exits **ATAS**.
+
+## 2. Design
+This section will give a high-level overview of how various components in **ATAS** function and interact with each other.
+
+### 2.1. Architecture
+![overall architecture](images/overall architecture.PNG)
+
+The architecture diagram above illustrates the high-level design of the **ATAS** application.  
+
+The `Atas` component contains all the other components in the application.  
+
+-   `Ui`: Reads user input, and shows the results of commands to the user
+
+-   `Logic`: Handles execution of user input commands
+
+-   `Model`: Stores the runtime data of the application
+
+-   `Storage`: Reads and stores data from a file stored on the user’s computer
+
+The following sections will explain each component in greater detail.
+
+### 2.2. UI Component
+1.  The `Ui` component reads user input which represents a command.
+
+2.  The `Ui` component shows the result of the command to the user.
+
+### 2.3. Logic Component
+
+The Logic component comprises the `Parser`, `Command`, and `CommandResult` classes:  
+
+1.  `Parser` will interpret the user command and return the corresponding `Command` object.  
+
+2.  `Command#execute()` is called to run the command, returning a `CommandResult` object.  
+
+3.  The `CommandResult` object will contain the output to be shown to the user.
+
+### 2.4. Model Component
+The Model component contains the `Task` and `TaskList` classes, which store the user’s schedule.
+
+![TaskList and Tasks](images/TaskList Task class diagram.PNG)
+
+### 2.5. Storage Component
+![Storage Class Diagram](images/storage.PNG)
```

Comment:
Since you are abstracting out the attributes, maybe can remove the attributes compartment? Will make the diagram more concise.